plugins/core/Standard/qCSF/include/qCSFCommands.h

```diff
@@ -0,0 +1,270 @@
+				bool conv = false;
+				QString scene = cmd.arguments().takeFirst();
+				if (scene == COMMAND_CSF_SCENE_SLOPE) {
+					cmd.print("Set scene to steep slope");
+					csf_rigidness = 1;
+				}
+				else if (scene == COMMAND_CSF_SCENE_RELIEF) {
+					cmd.print("Set scene to relief");
+					csf_rigidness = 2;
+				}
+				else if (scene == COMMAND_CSF_SCENE_FLAT) {
+					cmd.print("Set scene to flat");
+					csf_rigidness = 3;
+				}
+				else {
+					cmd.error("Unknown scene parameter. Defaulting to relief");
+					csf_rigidness = 2;
+				}
+			}else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_PROC_SLOPE)) 
+			{
+				cmd.arguments().pop_front();
+				cmd.print("Slope processing turned on");
+				csf_postprocessing = true;
+			}
+			else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_CLOTH_RESOLUTION)) {
+				cmd.arguments().pop_front();
+				bool conv = false;
+				cloth_resolution = cmd.arguments().takeFirst().toDouble(&conv);
+				if (!conv) {
+					return cmd.error(QObject::tr("Invalid parameter: value after \"-%1\"").arg(COMMAND_CSF_CLOTH_RESOLUTION));
+				}
+				cmd.print(QString("Custom cloth resulution set: %1").arg(cloth_resolution));
+			}
+			else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_MAX_ITERATION)) {
+				cmd.arguments().pop_front();
+				bool conv = false;
+				MaxIteration = cmd.arguments().takeFirst().toInt(&conv);
+				if (!conv) {
+					return cmd.error(QObject::tr("Invalid parameter: value after \"-%1\"").arg(COMMAND_CSF_MAX_ITERATION));
+				}
+				cmd.print(QString("Custom max iteration set: %").arg(MaxIteration));
+			}
+			else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_CLASS_THRESHOLD)) {
+				cmd.arguments().pop_front();
+				bool conv = false;
+				class_threshold = cmd.arguments().takeFirst().toDouble(&conv);
+				if (!conv) {
+					return cmd.error(QObject::tr("Invalid parameter: value after \"-%1\"").arg(COMMAND_CSF_CLASS_THRESHOLD));
+				}
+				cmd.print(QString("Custom class threshold set: %1").arg(class_threshold));
+			}
+			else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_EXPORT_GROUND))
+			{
+				cmd.arguments().pop_front();
+				cmd.print("Ground will be exported");
+				ExportGround = true;
+			}
+			else if (ccCommandLineInterface::IsCommand(arg, COMMAND_CSF_EXPORT_OFFGROUND))
+			{
+				cmd.arguments().pop_front();
+				cmd.print("Off-ground will be exported");
+				ExportOffground = true;
+			}
+			else
+			{
+				cmd.print("Set all parameters");
+				break;
+			}
+		}
+		
+		//instantiation a CSF class
+		CSF csf(csfPC);
+
+		//setup paramter
+		csf.params.k_nearest_points = 1;
+		csf.params.bSloopSmooth = csf_postprocessing;
+		csf.params.time_step = 0.65;
+		csf.params.class_threshold = class_threshold;
+		csf.params.cloth_resolution = cloth_resolution;
+		csf.params.rigidness = csf_rigidness;
+		csf.params.iterations = MaxIteration;
+
+		std::vector<int> groundIndexes;
+		std::vector<int> offGroundIndexes;
+		ccMesh* clothMesh = nullptr;
```

Comment:
it's a little bit misleading to declare this clothMesh variable here, as well as ExportClothMesh . ExportClothMesh is always false for now, but if it was turned to true, it wouldn't work either (as clothMesh is not managed later).